x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/api/util/tags.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { StoredRuleMigration } from '../../types';
+
+const vendorCodeMap: Record<StoredRuleMigration['original_rule']['vendor'], string> = {
+  splunk: 'SPL',
+};
+
+export const getTagsByVendor = (
+  vendor: StoredRuleMigration['original_rule']['vendor']
+): string[] => {
+  if (!(vendor in vendorCodeMap)) {
+    throw new Error(`Unsupported vendor for tagging rules: ${vendor}`);
+  }
+  return [`Migrated from ${vendorCodeMap[vendor]}`];
+};
```

Comment:
I think it's better to just use the (capitalized) vendor instead of the query language:
```
Migrated from Splunk
```

But if product/design wants that, we already have it defined in the `original_rule.query_language`, no need to create the map.

Another option is to combine both things:
```
Migrated from Splunk (SPL)
```

code:
```suggestion
export const getVendorTag = ( originalRule: OriginalRule ): string => {
  const { vendor, query_language: lang } = originalRule;
  const capitalizedVendor = `${vendor.charAt(0).toUpperCase()}${vendor.slice(1)}`;
  return `Migrated from ${capitalizedVendor} (${lang.toUpperCase()})`;
};
```